library/Freckle/App.hs

```diff
@@ -135,23 +142,62 @@
 --
 module Freckle.App
   ( runApp
+  , setLineBuffering
   , module X
+  , AppT(..)
+  , runAppT
   ) where
 
-import Prelude
+import Freckle.App.Prelude
 
 import Blammo.Logging as X
+import Control.Monad.Catch (MonadCatch, MonadThrow)
+import Control.Monad.IO.Unlift (MonadUnliftIO(..))
 import Control.Monad.Reader as X
+import Control.Monad.Trans.Resource (MonadResource, ResourceT, runResourceT)
 import Freckle.App.Database as X
 import System.IO (BufferMode(..), hSetBuffering, stderr, stdout)
 
 runApp
   :: HasLogger app
   => (forall b . (app -> IO b) -> IO b)
-  -> ReaderT app (LoggingT IO) a
+  -> AppT app IO a
```

Comment:
Even though this changes, users of `runApp` shouldn't have to change anything, as long as they were writing their actions against `MonadReader` and other constraints, and calling `runApp` without a type-signature on the action.